docs/proposals/algorithms/[OSPP]GAN and Self-taught Learning/[OSPP]A Proposal of Integrating GAN and Self-taught Learning to ianvs.md

```diff
@@ -0,0 +1,33 @@
+# Integrate GAN and Self-taught Learning into ianvs Lifelong Learning to Handle Unknown Tasks
+
+## Motivation
+
+In the process of ianvs lifelong learning, there would be a chance to confront unknown tasks, whose data are always heterogeneous small sample. Generate Adversarial Networks(GAN) is the start-of-art generative model and GAN can generate fake data according to the distribution of the real data. Naturally, we try to utilize GAN to handle small sample problem. Self-taught learning is an approach to improve classfication performance using sparse coding to construct higher-level features with the unlabeled data. Hence, we combine GAN and self-taught learning to help ianvs lifelong learning handle unknown tasks.
+
+### Goals
+
+* Handle unknown tasks
+* Implement of a lightweight GAN to solve small sample problem
+* Utilize self-taught learning to solve heterogeneous problem
+
+## Proposal
+We focus on the process of handling unknown tasks.     
+
+The overview is as follows:
+
+![](images/overview.png)
```

Comment:
A archtecture connecting ianvs lifelong learning and the proposed modules is needed.